build/config/compiler/BUILD.gn

```diff
@@ -158,6 +158,8 @@ config("disabled_warnings") {
       "-Wno-psabi",
       "-Wno-cast-function-type",
     ]
+  } else {
+    cflags += [ "-Wno-unknown-warning-option" ]
```

Comment:
This seems a larger global change.Is this required?

Since the above cflags already sets it, is this setting it again?

Reply:
It is not needed and I removed it local. I wanted to push but after rebase the lock application is not working anymore. I'm investigating and will push in this PR when fixed.

Reply:
@andy31415 I removed that change and re-enable all warnings that were disabled for Mbed OS builds.